### PR TITLE
Take care of email already exist in db case and fix user creation endpoint

### PR DESCRIPTION
**Pull request by inkhey**
- [X] Fix user creation endpoint (tests were missing and do not work)
- [X] handle properly "user email already exist" case + tests (tracim/tracim#793)
- [X] adapt user creation endpoint to "invitation" case (active/disable notification + allow autogenerated password if no password provided) (tracim/tracim#613)

### CODE REVIEW

File: backend/tracim_backend/lib/core/user.py [hunk 1]
```diff
@@ -271,7 +272,33 @@ def set_email(
         )
         return user
 
-    def _check_email(self, email: str) -> bool:
+    def _check_email(self, email) -> bool:
```
Comment (buxx) [inline backend/tracim_backend/lib/core/user.py]: typing of `email` ?

File: backend/tracim_backend/lib/core/user.py [hunk 2]
```diff
@@ -271,7 +272,33 @@ def set_email(
...
+            raise EmailAlreadyExistInDb(
+                'Email given {} already exist, please choose something else'.format(email)  # nopep8
+            )
+        return True
+
+    def check_email_already_in_db(self, email: str) -> bool:
+        """
+        Verify if given email does not already exist in db
+        """
+        return self._session.query(User.email).filter(User.email==email).all() != []  # nopep8
```
Comment (buxx) [inline backend/tracim_backend/lib/core/user.py]: For information: prefer use `.count()` for readability.

File: backend/tracim_backend/models/context_models.py [hunk 3]
```diff
@@ -99,17 +101,17 @@ class UserCreation(object):
...
+            timezone: str = None,
+            profile: str = None,
+            email_notification: bool = True,
     ) -> None:
         self.email = email
-        self.password = password
-        self.public_name = public_name
-        self.timezone = timezone
-        self.profile = profile
+        self.password = password or password_generator()
```
Comment (buxx) [inline backend/tracim_backend/models/context_models.py]: How is used this object ? What is set in `password` attribute ?

Comment (inkhey) [reply backend/tracim_backend/models/context_models.py]: this object is used in user creation endpoint, password is cleartext password. If admin set password, password is the password given by admin, else an autogenerated password is created for user (password_generator() function)

Comment (buxx) [reply backend/tracim_backend/models/context_models.py]: And these data are sent as response ? My reflexion is: "Is that a good idea to give clear text password is response ?".

Comment (lebouquetin) [reply backend/tracim_backend/models/context_models.py]: **never** send passwords (clear or hashed)

File: backend/tracim_backend/lib/core/user.py (hunk 2 above)
Comment (inkhey) [reply backend/tracim_backend/lib/core/user.py]: Done.

File: backend/tracim_backend/lib/core/user.py (hunk 1 above)
Comment (inkhey) [reply backend/tracim_backend/lib/core/user.py]: done.

File: backend/tracim_backend/models/context_models.py (hunk 3 above)
Comment (inkhey) [reply backend/tracim_backend/models/context_models.py]: This is not response, this is user creation request.

Comment (inkhey) [reply backend/tracim_backend/models/context_models.py]: when user do a user creation request, this object is created with user parameters, password_generator() create a default password for user, which can be send directly by email if notification is activated according to old password notification of tracimv1 (we probably should change this mecanism but actually this is not yet done).

Comment (buxx) [reply backend/tracim_backend/models/context_models.py]: Okay perfect.

**Pull request merged**